python/tk_desktop/project_commands_model.py

```diff
@@ -318,12 +315,9 @@ def __initialize_recents(self):
         # and a boolean saying whether the corresponding command has been registered
         self.__recents = {}
 
-        # need to know what login to find events for
-        login = ShotgunLogin.get_instance_for_namespace("tk-desktop").get_login()
-
         # pull down matching invents for the current project for the current user
         filters = [
-            ["user", "is", login],
+            ["user", "is", login.get_current_user(sgtk.platform.current_engine().tank)],
```

Comment:
what if `shotgun.yml` contains script credentials? Does that mean that it will use those, and the login core hook will then in turn be used? And does that mean that this may be a regression compared with the previous behaviour?

The previous code pulls the login out of the login framework? Should the new code follow the same pattern and pull the login info out of the shotgun_authentication module rather than sgtk?